web/src/components/storage/ProposalPage.jsx

```diff
@@ -216,6 +216,13 @@ export default function ProposalPage() {
     calculate(newSettings).catch(console.error);
   };
 
+  /**
+   * @todo Enable type checking and ensure the components are called with the correct props.
+   *
+   * @note The default value for `settings` should be `undefined` instead of an empty object, and
```

Comment:
NP: if so, we should keep the "?." operator.n

Reply:
Right now it cannot be undefined. If we initialize settings to undefined, then I would change the rest of the code a bit instead of having `.?` everywhere.